Content.Server/Speech/EntitySystems/EmoteBlockerSystem.cs

```diff
@@ -0,0 +1,89 @@
+using System.Collections.Frozen;
+using System.Linq;
+using Content.Server.Chat.Systems;
+using Content.Server.EntityEffects.Effects;
+using Content.Server.Popups;
+using Content.Server.Speech.Components;
+using Content.Server.Speech.EntitySystems;
+using Content.Shared.Actions;
+using Content.Shared.Chat.Prototypes;
+using Content.Shared.Emoting;
+using Content.Shared.EntityTable.EntitySelectors;
+using Content.Shared.Inventory;
+using Content.Shared.Speech;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Speech;
+
+public sealed class EmoteBlockerSystem : EntitySystem
+{
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    // Cache the Scream Emote's prototype so that it can be used when we deal with the Scream Action.
+    private EmotePrototype? _screamPrototype = null;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        // Intercept SceamActionEvent specifically because it's kinda like an emote, but isn't handled by the blocking code for emotes.
+        SubscribeLocalEvent<ScreamActionEvent>(OnScreamAction, before: [typeof(VocalSystem)]);
+        SubscribeLocalEvent<EmoteBlockerComponent, GetEmoteBlockersEvent>(OnGetEmoteBlockers);
+        SubscribeLocalEvent<EmoteBlockerComponent, InventoryRelayedEvent<GetEmoteBlockersEvent>>(OnRelayedGetEmoteBlockers);
+
+        SubscribeLocalEvent<PrototypesReloadedEventArgs>(OnPrototypeReload);
+    }
+
+    private void OnScreamAction(ScreamActionEvent args)
```

Comment:
This is cursed but l can't think of a better way, what you have here seems fine.